### PR TITLE
Header work

### LINKED ISSUE #76: findprovs returns object or string
using the method `ipfs.dht.findprovs` in node environment does not work like one would expect. Using the following code:
```
var ipfsAPI = require('ipfs-api')

var ipfs = ipfsAPI('localhost', '5001')

const hash = "QmXFmCEhamPHvpbzQ3HMFABrpAXgE7D9jzp6QtvyC8Xp55"

ipfs.dht.findprovs(hash, (err, res) => {
	if(err) throw err
	console.log(res)
	console.log(typeof res)
	console.log("Streamable?", !!res.readable)
	console.log('So what to do with `res` here?')
})
```

I get the following output sometimes (which works correctly):
```
object
Streamable? false
So what to do with `res` here?
```
but other times, it returns this (which is neither streamable or valid json)
```
string
Streamable? false
So what to do with `res` here?
```

Originally open issue https://github.com/ipfs/go-ipfs/issues/1851 on go-ipfs but seems to be a bug in this library.

### PR DESCRIPTION
**Pull request by Dignifiedquire**
Goal: Fix #76 still not there though..

When https://github.com/ipfs/go-ipfs/issues/1979 is merged we can remove the config settings in the daemon setup

### CODE REVIEW

File: src/request-api.js [hunk 1]
```diff
@@ -33,24 +34,18 @@ function onRes (buffer, cb) {
 
     if (chunkedObjects) {
       const parsed = []
-      res.on('data', chunk => parsed.push(JSON.parse(chunk)))
+      res.on('data', chunk => {
+        try {
+          parsed.push(JSON.parse(chunk))
+        } catch (err) {
+          // Browser quirks emit more than needed sometimes
```
Comment (diasdavid) [inline src/request-api.js]: `Emit more than needed`? What does this exactly mean? :)

File: tasks/daemons.js [hunk 2]
```diff
@@ -57,7 +57,14 @@ function startDisposableDaemons (callback) {
             throw err
           }
 
-          ipfsNodes[key].setConfig('API', '{"HTTPHeaders": {"Access-Control-Allow-Origin": ["*"]}}', err => {
+          const headers = {
+            HTTPHeaders: {
```
Comment (diasdavid) [inline tasks/daemons.js]: Let's put `When ipfs/go-ipfs#1979 is merged we can remove the config settings in the daemon setup` as a TODO here and a issue pointing to it

**Comment by diasdavid** [comment]
LGTM :)

File: src/request-api.js (hunk 1 above)
Comment (Dignifiedquire) [inline src/request-api.js]: if you have 3 chunks, you sometimes get: 
* chunk1
* chunk2
* chunk3
* chunk1+chunk2 => not parsable

do not ask me why, browsers are a weird bunch..

File: tasks/daemons.js (hunk 2 above)
Comment (Dignifiedquire) [inline tasks/daemons.js]: agreed

File: src/request-api.js (hunk 1 above)
Comment (diasdavid) [inline src/request-api.js]: What I'm thinking is that "stream" and "chunked" should not exist at the same time. "chunked" is a stream made of small parts of the same files, they should not be parsed individually. "stream" in the another hand, it is the ndjson stuff :)

Comment (Dignifiedquire) [inline src/request-api.js]: Correct, but "chunked" is represented as a (node) stream where each individual emitted event is equal to one chunk, that's why we are parsing it this way

**Pull request merged**